File: euler_p2.rb
#Problem 2 asks for the sum of all even terms in the Fibonacci sequence < 4 million
#we'll use the formula first_number + second_number = new_number,
#then reuse second_number as the next first_number and new_number as the next B
  my_sum = 0 #will keep the running total sum
  upper_limit = 4000000 #we will check numbers below this value
  first_number = 1 #this will hold the first number
  second_number = 2 #this is the second number
  new_number = 0 #this will hold the result
  puts my_sum
  while second_number <= upper_limit #once second_number >= our upper limit, we're done
     new_number = first_number + second_number #calculates the next number in the sequence
     puts "#{first_number} + #{second_number} = #{new_number}"
     if second_number % 2 == 0 #checks for an even number
       my_sum += second_number # If second_number is even, add it to our total sum
       puts "#{second_number} was even! my_sum is now: #{my_sum}"
     end
     #set up for next iteration
     first_number = second_number
     second_number = new_number
  end
  puts "Total sum: #{my_sum}" #outputs the sum of all even terms

File: eulerP2.js
//Problem 2 asks for the sum of all even terms in the Fibonacci sequence < 4 million
//we'll use the formula A+B=C, then reuse B as the next A and C as the next B
function main(){
  var mySum = 0;//will keep the running total sum
  var upperLimit = 4000000;//we will check numbers below this value
  var a = 1;//this will hold the first number
  var b = 2;//this is the second number
  var c = 0;//this will hold the result
  $('.output').append('<p>'+mySum+'</p>');
  while (b <= upperLimit){//once b >= our upper limit, we're done
     c = a + b;//calculates the next number in the sequence
     $('.output').append('<p>'+a+'+'+b+'='+c+'</p>');
     if(b % 2 == 0 ){//checks for an even number
       mySum += b;// If b is even, add it to our total sum
       $('.output').append('<p>b was even! mySum is now: '+mySum+'</p>');
     }
     //set up for next iteration
     a = b;
     b = c;
  }
  $('.output').append('<p>Total sum: '+mySum+'</p>');//outputs the sum of all even terms
}
$(document).ready(main);
